src/caffe/layers/accuracy_layer.cpp

```diff
@@ -50,11 +54,14 @@ void AccuracyLayer<Dtype>::Forward_cpu(const vector<Blob<Dtype>*>& bottom,
   const int num_labels = bottom[0]->shape(label_axis_);
   vector<Dtype> maxval(top_k_+1);
   vector<int> max_id(top_k_+1);
+  vector<Dtype> accuracies(dim, 0);
+  vector<Dtype> nums(dim, 0);
```

Comment:
Why?
They are usually small and it doesn't cost much to create them.
Also, they do not represent a state of the class so I don't think they should be class variables.